utils/loggers/comet/__init__.py

```diff
@@ -0,0 +1,503 @@
+    with Comet
+    """
+
+    def __init__(
+        self, opt, hyp, run_id=None, job_type="Training", **experiment_kwargs
+    ) -> None:
+        self.job_type = job_type
+        self.opt = opt
+        self.hyp = hyp
+
+        # Comet Flags
+        self.comet_mode = self.opt.comet_mode if self.opt.comet_mode else COMET_MODE
+
+        self.save_model = (
+            opt.comet_save_model if opt.comet_save_model else COMET_SAVE_MODEL
+        )
+        self.model_name = (
+            opt.comet_model_name if opt.comet_model_name else COMET_MODEL_NAME
+        )
+        self.overwrite_checkpoints = (
+            opt.comet_overwrite_checkpoints
+            if opt.comet_overwrite_checkpoints
+            else COMET_OVERWRITE_CHECKPOINTS
+        )
+
+        # Batch Logging Settings
+        self.log_batch_metrics = (
+            opt.comet_log_batch_metrics
+            if opt.comet_log_batch_metrics
+            else COMET_LOG_BATCH_METRICS
+        )
+        self.comet_log_batch_interval = (
+            opt.comet_log_batch_interval
+            if opt.comet_log_batch_interval
+            else COMET_BATCH_LOGGING_INTERVAL
+        )
+
+        # Dataset Artifact Settings
+        self.upload_dataset = (
+            self.opt.comet_upload_dataset
+            if self.opt.comet_upload_dataset
+            else COMET_UPLOAD_DATASET
+        )
+        self.resume = self.opt.resume
+
+        # Default parameters to pass to Experiment objects
+        self.default_experiment_kwargs = {
+            "log_code": False,
+            "log_env_gpu": True,
+            "log_env_cpu": True,
+            "project_name": COMET_PROJECT_NAME,
+        }
+        self.default_experiment_kwargs.update(experiment_kwargs)
+        self.experiment = self._get_experiment(self.comet_mode, run_id)
+
+        if self.opt.comet_artifact:
+            self.data_dict = self.download_dataset_artifact(self.opt.comet_artifact)
+
+        else:
+            with open(self.opt.data, errors="ignore") as f:
+                self.data_dict = self.process_data_dict(yaml.safe_load(f))
+
+        self.class_names = self.data_dict["names"]
+        self.num_classes = self.data_dict["nc"]
+
+        self.logged_images_count = 0
+        self.max_images = (
+            self.opt.comet_max_image_uploads
+            if self.opt.comet_max_image_uploads
+            else COMET_MAX_IMAGE_UPLOADS
+        )
+        if self.experiment is not None:
+            if run_id is None:
+                self.experiment.log_other("Created from", "YOLOv5")
+
+                if not isinstance(self.experiment, comet_ml.OfflineExperiment):
+                    workspace, project_name, experiment_id = self.experiment.url.split(
+                        "/"
+                    )[-3:]
+                    self.experiment.log_other(
+                        "Run Path",
+                        f"{workspace}/{project_name}/{experiment_id}",
+                    )
```

Comment:
Why do we need to log the Experiment Run Path? If the experiment is copied to another project / workspace, that value is gonna be outdated

Reply:
This is logged as a way to conveniently resume an experiment or use weights from an experiment 

Resuming
```
python train.py --resume "comet://<run path>"
``` 

Using Weights
```
python train.py --weights "comet://<run path>"
```